### PR TITLE
Fix extract_filename function

### PR DESCRIPTION
**Pull request by anutosh491**
If the `extract_filename` function is going to be editing `argv`, then it also needs to make a corresponding change to `argc`. Can lead to inconsistencies otherwise.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -60,19 +60,19 @@ bool should_print_version(int argc, char* argv[])
     return false;
 }
 
-std::string extract_filename(int argc, char* argv[])
+std::string extract_filename(int *argc, char* argv[])
```
Comment (JohanMabille) [inline src/main.cpp:63]: Prefer passing references instead of pointers when you can (which is the case for argc, not for argv)

**Review by JohanMabille — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 1 above)
Comment (anutosh491) [reply src/main.cpp]: Argh argh argh yes fundamental mistake. Overlooked it. Thanks !

Comment (vgvassilev) [reply src/main.cpp]: Can we add a test for this?

Comment (anutosh491) [reply src/main.cpp]: Hmm not sure, how we could test this out :(
Just something I noticed which I thought I'll address!

Comment (JohanMabille) [reply src/main.cpp]: You can forward declare the function in the test file, the definition will be found when linking.

**Review by JohanMabille — APPROVED** [review]

File: src/main.cpp (hunk 1 above)
Comment (anutosh491) [reply src/main.cpp]: Yeah that makes sense. Done.

File: test/test_interpreter.cpp [hunk 2]
```diff
@@ -9,6 +9,9 @@
 #include "doctest/doctest.h"
 #include "xeus-cpp/xinterpreter.hpp"
 
+// Forward declaration of extract_filename
+std::string extract_filename(int &argc, char* argv[]);
```
Comment (vgvassilev) [inline test/test_interpreter.cpp:13]: Now we need to call this function with different arguments and make sure it does what’s expected.

Comment (anutosh491) [reply test/test_interpreter.cpp:13]: Okay I'll add a basic test from what comes to my mind.

File: src/xutils.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                break;
+            }
+        }
+        return res;
+    }
+
+    interpreter_ptr build_interpreter(int argc, char** argv)
+    {
+        int interpreter_argc = argc;  // + 1;                                               // ...
+        const char** interpreter_argv = new const char*[interpreter_argc];
```
[the remark this answers is not in the document]
Comment (anutosh491) [reply src/xutils.cpp:86]: Cool I think we can look into any more refactoring through subsequent PRs. The PR has already deivated a bit from what it intended to do. I'll raise one to address the `build_interpreter` function and relevant tests next.

[the remark this answers is not in the document]
Comment (vgvassilev) [reply src/xutils.cpp:86]: Fair enough.

**Pull request merged**